tests/Validation/ValidationValidatorTest.php

```diff
@@ -1153,7 +1153,7 @@ public function testValidateSame()
         $this->assertFalse($v->passes());
 
         $v = new Validator($trans, ['foo' => 'bar'], ['foo' => 'Same:baz']);
-        $this->assertFalse($v->passes());
+        $this->assertTrue($v->passes());
```

Comment:
This change seems incorrect. From the docs for the `same` rule:

> The given field must match the field under validation.

The `foo` field value does not match the `baz` field value nor is that possible since there is no `baz` field in the validated payload at all.

Reply:
Yes, there is my point. This conflicts with "sometimes" rule. The current way we can't use *"same"* or *"different"* rule with *"sometimes"*.

My point is that this should only validate if both are present in the request

Reply:
Hmm, just read your issue and yes I see how the behavior is weird for the `different` rule. But for the `same` rule the behavior is exactly as expected and the above assertion should not be changed. If `foo` is in the payload and `baz` isn't then those values obviously aren't the same as `foo` value is `bar` and `bar !== null`.

Reply:
Yes, I actually got this behavior when using the *different* rule and then I tried to apply to the *same*. The behavior with the *different* is really weird. 

I see you point and I agree in parts. The point that made me think about the *same* rule is the behavior with the *sometimes* as well. I can't use the *same* with *sometimes* if only one is present in the request. The error message will be something like *"he both must match". But only one is present in request.

Reply:
You are not supposed to use `sometimes` with the `same` rule if the default behavior is not what you need. What you need is the ability to [conditionally add rules](https://laravel.com/docs/6.x/validation#conditionally-adding-rules) which exists for this precise reason:

```php
$validator->sometimes('foo', 'same', function ($input) {
    return $input->baz !== null;
});
```

You can do this for the `different` rule too and IMO this makes this PR redundant.

Reply:
This sounds a little more weird to me. The fact that Laravel is trying to validate fields that aren't present in the request.

I mean, this is one of many ways of solving this, but, is this right?

What I propose is to only validate fields that are in the request. If the field isn't present it isn't equal **and** it isn't different as well, it's undefined IMO.

What validator is doing is: 
```php
"something" == undefided? true|false (bool)
```
IMO this should be:

```php
"something" == undefided? undefined 
```

Reply:
We have two different topics here, how the `same` rule works/should work and your usecase in combination with the `sometimes` rule.

You've literally changed the behavior of the `same` rule to say that something is the same as nothing which is simply incorrect, just because it's convenient for your usecase. On top of that you are basically saying that this is a bugfix.

The `same` rule as well as its behavior and the test/assertion that you've changed exist since the first commit ever to the framework (2013) on GitHub. The assertion is proof that the author wanted it to behave that way, it's not a bug. Changing its behavior would IMO be a major breaking change which would affect literally every Laravel application out there.

> What I propose is to only validate fields that are in the request

This is not how the validator behaves at the moment and like I said changing this would be a major BC. The behavior of the validator is very well documented - the very first sentence in the documentation link above about conditionally adding rules is:

> In some situations, you may wish to run validation checks against a field only if that field is present in the input array

This clearly implies that the validation is run by default against all the specified fields even if the values are not present in the input and that the authors of the framework have added this feature for the exact usecase that you have here.

Reply:
I just said that I tried the same behavior that I had with *different* with the *same*. But no problems, I can undo my modifications to *same* and let only the *different*. No problems about it.

My problem was originally with *different* and the usecase is in the issue

Reply:
But this is just a proposal to the issue that I opened. If the *same* (or even this entire PR) is not relevant I can modify or even close, no problems about it. Like I said, I had this weird behavior with the *different* rule, and I think really weird this behavior with the *different* when using sometimes because it almost says that something that is in request is equal to something that isn't, making validations fails

Reply:
Yeah, I understand. The assert/behavior you should change (if the maintainers feel like the change is warranted) is this:

```php
$v = new Validator($trans, ['foo' => 'bar'], ['foo' => 'Different:baz']);
$this->assertFalse($v->passes());
```

`bar !== null` so the validation should IMO actually pass, not fail.

The behavior of `same` should remain the same.

Reply:
Yeah, agreed. I'll undo the changes applied to the *same* rule. Tks